### ISSUE TITLE
Scheduled messages not being delivered

### ISSUE THREAD

**Issue by Lu-vam** [instedd/aida#125]
Hi! When the bot is connected to Facebook Messenger, surveys and scheduled messages (except from 'scheduled outgoing messages after last message received') are not delivered to the users . This happens even if the bot has been approved by Facebook (including messaging subscribtions) and even if users have been interacting with the bot in the past minutes / hours.
Thanks for looking into this problem.

**Comment by mmuller** [comment]
Hi @Lu-vam we couldn't reproduce this issue.
Could you provide us with the Bot ID?

**Comment by Lu-vam** [comment]
https://aida-stg.instedd.org/b/18

**Comment by mmuller** [comment]
@Lu-vam It should be working now. Could you try again please?

**Comment by nditada** [comment]
@Lu-vam You mentioned that scheduled messages are not going out even for users that just recently talked to the bot. Is that something that happened on the bot Id above?

**Comment by Lu-vam** [comment]
Yes that's correct!

**Comment by Lu-vam** [comment]
From the data in the API looks like the scheduled messages where sent out multiple times - however they were never delivered to users on Messenger.

**Comment by Lu-vam** [comment]
@nditada @mmuller a similar problem still happens also with surveys - the bot does not send them out to all users!

**Comment by waj** [comment]
@Lu-vam I just deployed a new version in staging (aida-stg.instedd.org) with a fix that might be related to that issue. Could you please try to schedule the survey again?

**Comment by Lu-vam** [comment]
Good morning @waj! We are getting there..... as you can see from the picture attached, this time the survey was sent to 5 users (more people than before), but still bot to everyone! Could it be due to some Facebook restrictions?

![image](https://user-images.githubusercontent.com/36440133/37699418-c7dda138-2ce7-11e8-8099-9a657e87b32f.png)

**Comment by Lu-vam** [comment]
[https://developers.facebook.com/docs/messenger-platform/policy/policy-overview#subscription_messaging](url)

"Once a Page is approved, any app subscribed to the Page may send subscription messages on the Page's behalf using the Send API's NON_PROMOTIONAL_SUBSCRIPTION message tag to anyone that begins a conversation with the Page."

**Comment by waj** [comment]
@Lu-vam The survey is not sent to users that didn't select the language yet. I checked the logs and there was sent to 6 users, for one of them Facebook replied with the error "(#10) This person isn't available right now". 

We must use the NON_PROMOTIONAL_SUBSCRIPTION tag to send messages outside the 24hs window

**Comment by Lu-vam** [comment]
Hey @waj that's true - great! Is the NON_PROMOTIONAL_SUBSCRIPTION tag included automatically by AIDA? Thanks

**Comment by waj** [comment]
@Lu-vam not right now, that's why surveys and scheduled messages cannot be sent to users that didn't interact recently with the bot

**Comment by nditada** [comment]
Will open a separate issue for configuring the "NON_PROMOTIONAL_SUBSCRIPTION" tag in the facebook channel for a bot.

**Comment by Lu-vam** [comment]
@nditada all our chatbots will need to have "NON_PROMOTIONAL_SUBSCRIPTION" capability! It could be a on/off feature. We can discuss this on the phone if needed!

**Comment by nditada** [comment]
@Lu-vam perfect: #140
